src/Symfony/Component/Notifier/Bridge/Smsapi/SmsapiTransport.php

```diff
@@ -84,6 +84,14 @@ protected function doSend(MessageInterface $message): SentMessage
             throw new TransportException(sprintf('Unable to send the SMS: "%s".', $content['message'] ?? 'unknown error'), $response);
         }
 
-        return new SentMessage($message, (string) $this);
+        $sentMessage = new SentMessage($message, (string) $this);
+
+        if (isset($content['list']) && \is_array($content['list'])) {
+            if ($smsStatus = current($content['list'])) {
+                $sentMessage->setMessageId($smsStatus['id'] ?? '');
```

Comment:
Do we really want to set an empty string? What about avoiding the method call in this scenario?

Reply:
My solution is rather an exaggerated caution. There should always be a message ID for a 200 response.
But I did it as [like here](https://github.com/symfony/symfony/blob/c2fd6b003a2b3bf2928b8f9d8bffdd3af22643b8/src/Symfony/Component/Notifier/Bridge/Smsc/SmscTransport.php#L91)
Today is no possibility to omit "id" in vendor response. Rather, it is a safeguard for the future. We can also solve this by omitting.